pyiron/atomistics/structure/visualization.py

```diff
@@ -0,0 +1,604 @@
+# coding: utf-8
+# Copyright (c) Max-Planck-Institut für Eisenforschung GmbH - Computational Materials Design (CM) Department
+# Distributed under the terms of "New BSD License", see the LICENSE file.
+
+import numpy as np
+import warnings
+from matplotlib.colors import rgb2hex
+from pyiron_base import Settings
+from scipy.interpolate import interp1d
+
+__author__ = "Joerg Neugebauer, Sudarsan Surendralal"
+__copyright__ = (
+    "Copyright 2020, Max-Planck-Institut für Eisenforschung GmbH - "
+    "Computational Materials Design (CM) Department"
+)
+__version__ = "1.0"
+__maintainer__ = "Sudarsan Surendralal"
+__email__ = "surendralal@mpie.de"
+__status__ = "production"
+__date__ = "Sep 1, 2017"
+
+s = Settings()
+
+
+def plot3d_plotly(
+    atoms,
+    scalar_field=None,
+    particle_size=1.0,
+    camera="orthographic",
+    view_plane=np.array([1, 1, 1]),
+    distance_from_camera=1.25,
+    opacity=1,
+):
+    """
+    Make a 3D plot of the atomic structure.
+
+    Args:
+        camera (str): 'perspective' or 'orthographic'. (Default is 'perspective'.)
+        particle_size (float): Size of the particles. (Default is 1.)
+        scalar_field (numpy.ndarray): Color each atom according to the array value (Default is None, use coloring
+            scheme.)
+        view_plane (numpy.ndarray): A Nx3-array (N = 1,2,3); the first 3d-component of the array specifies
+            which plane of the system to view (for example, [1, 0, 0], [1, 1, 0] or the [1, 1, 1] planes), the
+            second 3d-component (if specified, otherwise [1, 0, 0]) gives the horizontal direction, and the third
+            component (if specified) is the vertical component, which is ignored and calculated internally. The
+            orthonormality of the orientation is internally ensured, and therefore is not required in the function
+            call. (Default is np.array([0, 0, 1]), which is view normal to the x-y plane.)
+        distance_from_camera (float): Distance of the camera from the structure. Higher = farther away.
+            (Default is 14, which also seems to be the NGLView default value.)
+        opacity (float): opacity
+
+    Returns:
+        (plotly.express): The NGLView widget itself, which can be operated on further or viewed as-is.
+
+    """
+    try:
+        import plotly.express as px
+    except ImportError:
+        print("plotly not installed")
```

Comment:
Maybe a `ModuleNotFoundError` with a error message recommending `plot3d()` would be better than a print statement

Reply:
Oh is `ModuleNotFoundError` raised when it's not found?

Reply:
Yes. `ImportError` is raised if that package is there but the module you want to import isn't. Here if plotly is not there a `ModuleNotFoundError` is raised.

Reply:
Nice, I learned something new!